dev_tools/check_shellscript_set_options.py

```diff
@@ -12,22 +12,47 @@
     from pathlib import Path
 
 
-def _sets_options_or_is_nolint(line: str) -> bool:
-    return line.strip() in ["set -euxo pipefail", "# nolint(set_options)"]
+def _sets_options_or_is_nolint(line: str, expected_options: str) -> bool:
+    return line.strip() in [ expected_options, "# nolint(set_options)"]
 
 
-def _is_valid_bash_file(filename: Path) -> bool:
-    return any(_sets_options_or_is_nolint(line) for line in filename.open().readlines())
+def _is_valid_shell_file(filename: Path, expected_options: str) -> bool:
+        with filename.open() as fd:
+            return any(_sets_options_or_is_nolint(line, expected_options) for line in fd.readlines())
+
+
+def _separate_bash_from_sh_files(filenames: Sequence[Path]) -> tuple[list[Path], list[Path]]:
+    bash_files = []
+    sh_files = []
+    for filename in filenames:
+        first_line = filename.open().readline()
+        if first_line.startswith(("#!/bin/bash", "#!/usr/bin/bash")):
+            bash_files.append(filename)
+        elif first_line.startswith(("#!/bin/sh", "#!/usr/bin/sh")):
+            sh_files.append(filename)
+        else:
+            msg = f"Unknown shell in {filename}: {first_line}. Only use this hook in combination with 'check-executables-have-shebangs' from https://github.com/pre-commit/pre-commit-hooks"
+            raise ValueError(msg)
+
+    return bash_files, sh_files
+
+
+def _are_shell_files_valid(shell_files: list[Path], expected_options :str) -> None:
+    invalid_shell_files = [filename for filename in shell_files if not _is_valid_shell_file(filename, expected_options)]
+    for filename in invalid_shell_files:
+        print(f"Error: {filename} does not contain '{expected_options}'")
+
+    return bool(not invalid_shell_files)
 
 
 def main(argv: Sequence[str] | None = None) -> int:
     args = parse_arguments(argv)
 
-    invalid_bash_files = [filename for filename in args.filenames if not _is_valid_bash_file(filename)]
-    for filename in invalid_bash_files:
-        print(f"Error: {filename} does not contain 'set -euxo pipefail'")
+    bash_files, sh_files = _separate_bash_from_sh_files(args.filenames)
+    are_all_files_valid = _are_shell_files_valid(bash_files, "set -euxo pipefail")
+    are_all_files_valid &= _are_shell_files_valid(sh_files, "set -eux")
```

Comment:
```suggestion
    are_all_files_valid = _are_shell_files_valid(bash_files, "set -euxo pipefail") and  _are_shell_files_valid(sh_files, "set -eux")
```

Reply:
I'd like to avoid short-circuiting, and I'm doing that successfully with my code:

```python3
>>> def check(i):
...     print ("geeks")
...     return i
...
>>> a = True
>>> a &= check(True)
geeks
>>> a &= check(False)
geeks
>>> a &= check(True)
geeks
>>> b = check(False) and check(True)
geeks
>>> b = check(True) and check(False)
geeks
geeks
```